drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+	{ BIT(6), NULL },
+	{ BIT(7), NULL },
+	/* MSVW02 */
+	{ BIT(8), vw_host_rst_warn_isr },
+	{ BIT(9), NULL },
+	{ BIT(10), NULL },
+	{ BIT(11), NULL },
+	/* MSVW03 */
+	{ BIT(12), vw_sus_warn_isr },
+	{ BIT(13), NULL },
+	{ BIT(14), NULL },
+	{ BIT(15), NULL },
+};
+
+const struct espi_isr peripherals_isr[] = {
+	{ BIT(0), uart0_isr },
+	{ BIT(1), uart1_isr },
+	{ BIT(5), ibf_isr },
+	{ BIT(22), port80_isr },
+	{ BIT(23), port80_isr },
+};
+
+static u8_t bus_isr_cnt = sizeof(espi_bus_isr)/sizeof(struct espi_isr);
+static u8_t m2s_vwires_isr_cnt = sizeof(m2s_vwires_isr)/sizeof(struct espi_isr);
+static u8_t periph_isr_cnt = sizeof(peripherals_isr)/sizeof(struct espi_isr);
+
+static void espi_xec_bus_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct espi_xec_config *config = dev->config->config_info;
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(config->bus_girq_id);
+	REG32(MCHP_GIRQ_SRC_ADDR(config->bus_girq_id)) = girq_result;
+
+	for (int i = 0; i < bus_isr_cnt; i++) {
+		struct espi_isr entry = espi_bus_isr[i];
+
+		if (girq_result & entry.girq_bit) {
+			if (entry.the_isr != NULL) {
+				entry.the_isr(dev);
+			}
+		}
+	}
+}
+
+static void espi_xec_vw_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct espi_xec_config *config = dev->config->config_info;
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(config->vw_girq_id);
+	REG32(MCHP_GIRQ_SRC_ADDR(config->vw_girq_id)) = girq_result;
+
+	for (int i = 0; i < m2s_vwires_isr_cnt; i++) {
+		struct espi_isr entry = m2s_vwires_isr[i];
+
+		if (girq_result & entry.girq_bit) {
+			if (entry.the_isr != NULL) {
+				entry.the_isr(dev);
+			}
+		}
+	}
+}
+
+static void espi_xec_periph_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct espi_xec_config *config = dev->config->config_info;
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(config->pc_girq_id);
+	REG32(MCHP_GIRQ_SRC_ADDR(config->pc_girq_id)) = girq_result;
+
+	for (int i = 0; i < periph_isr_cnt; i++) {
+		struct espi_isr entry = peripherals_isr[i];
+
+		if (girq_result & entry.girq_bit) {
+			if (entry.the_isr != NULL) {
+				entry.the_isr(dev);
+			}
+		}
+	}
+}
+
+static int espi_xec_init(struct device *dev);
+
+static const struct espi_driver_api espi_xec_driver_api = {
+	.config = espi_xec_configure,
+	.get_channel_status = espi_xec_channel_ready,
+	.read_request = espi_xec_read_request,
+	.write_request = espi_xec_write_request,
+	.send_vwire = espi_xec_send_vwire,
+	.receive_vwire = espi_xec_receive_vwire,
+	.send_oob = espi_xec_send_oob,
+	.receive_oob = espi_xec_receive_oob,
+	.flash_read = espi_xec_flash_read,
+	.flash_write = espi_xec_flash_write,
+	.flash_erase = espi_xec_flash_erase,
+	.manage_callback = espi_xec_manage_callback,
+};
+
+static struct espi_xec_data espi_xec_data;
+static const struct espi_xec_config espi_xec_config = {
+	.base_addr = DT_MICROCHIP_XEC_ESPI_0_BASE_ADDRESS,
+	.bus_girq_id = MCHP_ESPI_IO_GIRQ,
```

Comment:
Same as BARs, IRQs should all come from DTS

Reply:
Fixed. Defined all group IRQs in DTS